hawc/apps/eco/models.py

```diff
@@ -97,48 +84,27 @@ class StudySetting(models.IntegerChoices):
         Ecoregion, blank=True, help_text="Select one or more Level III Ecoregions, if known",
     )
 
-    class HabitatType(models.IntegerChoices):
-        TERR = 0, "Terrestrial"
-        RIP = 1, "Riparian"
-        FRESH = 2, "Freshwater aquatic"
-        ESTU = 3, "Estuarine"
-        MAR = 4, "Marine"
-
-    habitat = models.IntegerField(
+    habitat_type = models.ForeignKey(
+        Vocab,
         verbose_name="Habitat",
-        choices=HabitatType.choices,
+        limit_choices_to=Q(category="Habitat")
+        & Q(parent__isnull=True),  # not sure if isnull is best approach here
+        on_delete=models.CASCADE,
         blank=True,
         help_text="Select the habitat to which the evidence applies",
+        related_name="+",
     )
 
```

Comment:
@shapiromatron am I headed in the right direction here? I created a Vocab model with category, value, and parent fields and am filtering it to replace the IntegerChoices fields I had originally.

Reply:
two options -

a.  we could have two categories, `habitat_type` and `habitat`. we can add additional validation to make sure that the a habitat's parent is the selected habitat type.
b. we could infer habitat_type from habitat without any data entry